src/components/DnsSettings.tsx

```diff
@@ -37,80 +49,124 @@ function getUpdatedSettings(settings: DnsSettings, action: Action): DnsSettings
   };
 }
 
-const DnsSettingsForm: FC<Props> = ({ settings, isEditor, onUpdate }) => {
-  const values = {
-    ...defaultValues,
-    ...settings.dns,
-  };
+const DnsSettingsForm: FC<Props> = ({ settings, isEditor, labels, onUpdate }) => {
+  const values = useMemo(
+    () => ({
+      ...defaultValues,
+      ...settings.dns,
+    }),
+    [settings.dns]
+  );
```

Comment:
As far as I can see, you only use `dns` in this component, so you don't need to provide the full `settings` object.

Reply:
Good call! It's a consequence of the state management of the parent component. I want to come back and clean that up via a switch to `Form` in a future PR